ykrt/src/compile/jitc_yk/aot_ir.rs

```diff
@@ -50,6 +53,29 @@ const LLVM_DEBUG_CALL_NAME: &str = "llvm.dbg.value";
 /// point call.
 const CTRL_POINT_ARGIDX_INPUTS: usize = 2;
 
+/// A struct identifying a line in a source file.
+#[derive(Debug)]
+#[deku_derive(DekuRead)]
+struct LineInfoLoc {
+    /// The path to the containing source file.
+    ///
+    /// The file existed at AOT compile time and there is no guarantee that it still exists now.
+    #[deku(until = "|v: &u8| *v == 0", map = "map_to_pathbuf")]
+    filename: PathBuf,
```

Comment:
I think this means that every `LineInfoLoc` has a pointer to a full C string on the heap? Isn't that going to be quite punishing? I imagine we can probably observe the slowdown that causes on deserialisation?

Reply:
Yep. I've not optimised anything at this point. Really we need a table with indices.

Reply:
Right, but how much does this slow things down by? I'm *generally* fine with not optimising things, but I worry this could massively slow down deserialisation, but maybe I'm worrying over nothing.

Reply:
Thinking about it, for a large program like yklua, this is probably going to be too expensive, so maybe I need to rethink this part.

Reply:
Please measure before making a decision.

Reply:
do you mean measure master (no debuginfo) against this branch (some debuginfo)? that's about all I can do without actually implementing the other way first.

Reply:
Yes, master vs this branch: we just need to know if this is a performance cost that we can measure.

Reply:
Surprisingly the difference isn't measurable for yklua at millisecond granularity. About 34ms before and after.

I measured just the call to the deserialiser like this:
```diff
diff --git a/ykrt/src/compile/jitc_yk/aot_ir.rs b/ykrt/src/compile/jitc_yk/aot_ir.rs
index 06b56f89..843aba02 100644
--- a/ykrt/src/compile/jitc_yk/aot_ir.rs
+++ b/ykrt/src/compile/jitc_yk/aot_ir.rs
@@ -257,7 +257,10 @@ impl std::fmt::Display for Module {

 /// Deserialise an AOT module from the slice `data`.
 pub(crate) fn deserialise_module(data: &[u8]) -> Result<Module, Box<dyn Error>> {
-    let ((_, _), modu) = Module::from_bytes((data, 0))?;
+    use std::hint::black_box;
+    let start = std::time::Instant::now();
+    let ((_, _), modu) = black_box(Module::from_bytes((data, 0)))?;
+    eprintln!("deser took: {}ms", start.elapsed().as_millis());
     Ok(modu)
 }
```

Do you see anything amiss, or do I believe my results?

Reply:
I don't know how this works, so I'm relying on you to tell me! But if in doubt, let's just do indexes.

Reply:
It's not hard to implement, so me may as well.

Reply:
Works for me.

Reply:
https://github.com/ykjit/yk/pull/1307/commits/3d0024920e936aeeef8c1fbaec6811c737663681 implemented indices.